### PR TITLE
[iOS] Update inner stack correctly when popped fast

### PR DESCRIPTION
**Pull request by rmarinho**
### Description of Change ###

When iOS chains animations on popping pages it was failing to update the inner stack of NavigationPage.

We change this to be updated when we are removed from the UINavigatioNController. 

### Bugs Fixed ###

https://bugzilla.xamarin.com/show_bug.cgi?id=39908

### PR Checklist ###

- [ ] Has tests (if omitted, state reason in description)
- [x] Rebased on top of master at time of PR
- [x] Changes adhere to coding standard
- [x] Consolidate commits as makes sense

### CODE REVIEW

File: Xamarin.Forms.Core/NavigationPage.cs [hunk 1]
```diff
@@ -251,6 +251,17 @@ protected override bool OnBackButtonPressed()
 
 			var page = (Page)InternalChildren.Last();
 
+			return await RemoveAsyncInner(page, animated, fast);
+		}
+
+		[EditorBrowsable(EditorBrowsableState.Never)]
+		public async Task<Page> RemoveAsyncInner(Page page, bool animated, bool fast)
```
Comment (StephaneDelcroix) [inline Xamarin.Forms.Core/NavigationPage.cs]: make it internal

**Comment by rmarinho** [comment]
build

File: Xamarin.Forms.Core/NavigationPage.cs (hunk 1 above)
Comment (samhouts) [reply Xamarin.Forms.Core/NavigationPage.cs]: Can't make internal, but maybe put this on `INavigationPageController` in `Internals` namespace and implement explicitly here?

File: Xamarin.Forms.Platform.iOS/Renderers/NavigationRenderer.cs [hunk 2]
```diff
@@ -991,13 +994,26 @@ public override bool ShouldAutorotateToInterfaceOrientation(UIInterfaceOrientati
 			}
 
 			public override bool ShouldAutomaticallyForwardRotationMethods => true;
+
+			public override async void DidMoveToParentViewController(UIViewController parent)
+			{
+				//we are being removed from the UINavigationPage
+				if (parent == null)
+				{
+					System.Diagnostics.Debug.WriteLine("being removed");
```
Comment (samhouts) [inline Xamarin.Forms.Platform.iOS/Renderers/NavigationRenderer.cs]: maybe remove this if we don't need it anymore

**Comment by rmarinho** [comment]
@samhouts and @StephaneDelcroix  address your comments , ready for other review.

**Comment by adrianknight89** [comment]
@rmarinho Will we see this merged soon?

**Comment by rmarinho** [comment]
@adrianknight89 can you review ? :)

**Comment by adrianknight89** [comment]
@rmarinho I can take a look at this and my 2 pending PRs, but I'm having some technical issues with VS2017. Can somebody on your end follow these steps to reproduce the same issues?

- Clean install Windows 10 Enterprise
- Install VS2017 Enterprise
- Check out Xamarin.Forms
- Notice that compilation throws many errors because the solution file cannot recognize .NET Standard 2.0.
- Observe that if you create a blank Console application, then the target framework is indeed 2.0. Also, CMDing `dotnet --info` shows Core 2 is installed.

That said, I think the project hasn't been properly migrated to the new architecture or I'm missing something I can't figure out. I've even created an Azure VM and installed VS2017 there. Same issues.

Also, does Xamarin.Forms still work on VS2015? I might need to install that on a separate machine for the time being.

File: docs/Xamarin.Forms.Core/Xamarin.Forms/NavigationPage.xml [hunk 3]
```diff
@@ -1080,5 +1080,37 @@ public class MyPage : NavigationPage
...
+        <Parameter Name="animated" Type="System.Boolean" />
+        <Parameter Name="fast" Type="System.Boolean" />
+      </Parameters>
+      <Docs>
+        <param name="page">To be added.</param>
+        <param name="animated">To be added.</param>
+        <param name="fast">To be added.</param>
+        <summary>To be added.</summary>
+        <returns>To be added.</returns>
+        <remarks>To be added.</remarks>
```
Comment (adrianknight89) [inline docs/Xamarin.Forms.Core/Xamarin.Forms/NavigationPage.xml]: Nothing for params, summary, returns, and remarks?

File: docs/Xamarin.Forms.Core/Xamarin.Forms/INavigationPageController.xml [hunk 4]
```diff
@@ -129,6 +129,30 @@
...
+        <Parameter Name="animated" Type="System.Boolean" />
+        <Parameter Name="fast" Type="System.Boolean" />
+      </Parameters>
+      <Docs>
+        <param name="page">To be added.</param>
+        <param name="animated">To be added.</param>
+        <param name="fast">To be added.</param>
+        <summary>To be added.</summary>
+        <returns>To be added.</returns>
+        <remarks>To be added.</remarks>
```
Comment (adrianknight89) [inline docs/Xamarin.Forms.Core/Xamarin.Forms/INavigationPageController.xml]: Same here.

Comment (rmarinho) [reply docs/Xamarin.Forms.Core/Xamarin.Forms/INavigationPageController.xml]: this is done by the Docs team

**Pull request merged**